### PR TITLE
MBS-6015 Track Gradle remote build cache errors

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
Track Gradle cache errors from clients.

`build.cache.errors.[load|store].<http status code>` - the new metric

See key points in my comments.

### CODE REVIEW

File: subprojects/gradle/build-metrics/src/gradleTest/kotlin/com/avito/android/plugin/build_metrics/cache/HttpBuildCacheMetricsTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import com.avito.android.plugin.build_metrics.statsdMetrics
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.Test
+import java.io.File
+
+internal class HttpBuildCacheMetricsTest : HttpBuildCacheTestFixture() {
+
```
Comment (eugene-krivobokov) [inline subprojects/gradle/build-metrics/src/gradleTest/kotlin/com/avito/android/plugin/build_metrics/cache/HttpBuildCacheMetricsTest.kt:11]: Test cases for the new metric

File: subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/SentryConsumer.kt [hunk 2]
```diff
@@ -1,97 +0,0 @@
...
-import com.avito.logger.Logger
-import com.avito.logger.LoggerFactory
-import com.avito.logger.create
-import io.sentry.connection.EventSendCallback
-import io.sentry.event.Event
-import org.gradle.BuildResult
-import org.gradle.api.Project
-
-// todo вынести в фабрику знания про project
-internal class SentryConsumer(
```
Comment (eugene-krivobokov) [inline subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/SentryConsumer.kt:18]: Not used for a long time. 
Maybe we will add a replacement later. I'll decide while troubleshooting network errors.

File: subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/SlowTasksListener.kt [hunk 3]
```diff
@@ -19,19 +17,6 @@ internal class SlowTasksListener(
             .flatMap { it.getTasks() }
             .filter { it.internalState.isActionable }
 
-        // TODO: filter only cacheable tasks after https://github.com/gradle/gradle/issues/9333 (MBS-7244)
```
Comment (eugene-krivobokov) [inline subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/SlowTasksListener.kt:22]: Found it in a `BuildCacheRemoteLoadBuildOperationType`. Will add in MBS-7244

File: subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/cache/BuildCacheOperationListener.kt [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import org.gradle.api.Project
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.provider.Provider
+import org.gradle.caching.http.HttpBuildCache
+import org.gradle.caching.internal.controller.operations.LoadOperationDetails
+import org.gradle.caching.internal.controller.operations.StoreOperationDetails
+import org.gradle.internal.operations.BuildOperationDescriptor
+import org.gradle.internal.operations.OperationFinishEvent
+
+internal class BuildCacheOperationListener(
```
Comment (eugene-krivobokov) [inline subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/cache/BuildCacheOperationListener.kt:20]: The logic for the new metric is here

File: subprojects/gradle/gradle-profile/src/main/kotlin/com/avito/android/gradle/metric/GradleCollector.kt [hunk 5]
```diff
@@ -37,22 +36,13 @@ class GradleCollector(
         result.gradle?.also { cleanup(it) }
     }
 
-    override fun onOutput(output: CharSequence?) {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/gradle-profile/src/main/kotlin/com/avito/android/gradle/metric/GradleCollector.kt:40]: Not needed anymore. 
We can find almost all info in build operations in a more type safe way.

**Review by dsvoronin — APPROVED** [review]

**Comment by eugene-krivobokov** [comment]
TIL: [org.gradle.unsafe.build-cache.remote-continue-on-error](https://github.com/gradle/gradle/pull/14492/files)

**Comment by dsvoronin** [comment]
> TIL: [org.gradle.unsafe.build-cache.remote-continue-on-error](https://github.com/gradle/gradle/pull/14492/files)

could be harmful if it's not a one-time error, but for example missing vpn connection

**Pull request merged**